File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
from collections import defaultdict
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        both = set()
        a_map = defaultdict(int)
        b_map = defaultdict(int)
        res   = []
        for num1,num2 in zip(A,B):
            a_map[num1] += 1
            b_map[num2] += 1
            if num1 in b_map:
                both.add(num1)
            if num2 in a_map:
                both.add(num2)
            res.append(len(both))
        return res

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.go
func findThePrefixCommonArray(A []int, B []int) []int {
    N := len(A)
    res := make([]int, N)

    a_map := make(map[int]int)
    b_map := make(map[int]int)

    var count = 0
    for i:=0; i < N; i++ {
        a_map[A[i]] += 1
        b_map[B[i]] += 1

        if A[i] !=  B[i]{
            count += a_map[B[i]]
            count += b_map[A[i]]
        }else {
            count += a_map[A[i]]
        }
        res[i] = count
    }
    return res
}